http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -197,27 +206,32 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
         machineConfig =
           if (AppType.Allowed == req.appType) machineConfigFromReqAndConfig.copy(autoscalingEnabled = true)
           else machineConfigFromReqAndConfig
+
         // We want to know if the user already has a nodepool with the requested config that can be re-used
         userNodepoolOpt <- nodepoolQuery
           .getMinimalByUserAndConfig(originatingUserEmail, cloudContext, machineConfig)
           .transaction
-
-        nodepool <- userNodepoolOpt match {
-          case Some(n) =>
-            log.info(ctx.loggingCtx)(
-              s"Reusing user's nodepool ${n.id} in ${saveClusterResult.minimalCluster.cloudContext.asStringWithProvider} with ${machineConfig}"
-            ) >> F.pure(n)
-          case None =>
-            for {
-              _ <- log.info(ctx.loggingCtx)(
-                s"No nodepool with ${machineConfig} found for this user in project ${saveClusterResult.minimalCluster.cloudContext.asStringWithProvider}. Will create a new nodepool."
-              )
-              saveNodepool <- F.fromEither(
-                getUserNodepool(clusterId, cloudContext, originatingUserEmail, machineConfig, ctx.now)
-              )
-              savedNodepool <- nodepoolQuery.saveForCluster(saveNodepool).transaction
-            } yield savedNodepool
-        }
+        nodepool <-
+          if (req.autopilot.isDefined) {
+            F.pure(saveClusterResult.defaultNodepool.toNodepool())
```

Comment:
just to be clear, this nodepool that's saved to the DB won't be truly accurate with what GCP is actually doing?

Reply:
yes..it's the same as how we do things with Azure...it'll be a big overhaul with DB schema if we don't want to do this because APP table has `nodepoolId` as FK and it doesn't have reference kubernetest_cluster table directly